app/src/main/java/com/example/drawingApp/ColorPickerUtility.kt

```diff
@@ -0,0 +1,222 @@
+            val green = Color.green(intColor)
+            val blue = Color.blue(intColor)
+
+            hView.forceText(hsv[0].toString())
+            sView.forceText((hsv[1] * 100).toString())
+            vView.forceText((hsv[2] * 100).toString())
+            rView.forceText(red.toString())
+            gView.forceText(green.toString())
+            bView.forceText(blue.toString())
+            chView.forceText(Integer.toHexString(Color.rgb(red, green, blue)).drop(2))
+
+            square.hue = hsv[0]
+            square.sat = hsv[1]
+            square.valu = hsv[2]
+            square.invalidate()
+
+            slider.sliderSetHue(hsv[0])
+
+            activeBox.setBackgroundColor(intColor)
+        }
+        //called whenever text needs to be updated
+        val updateForceViews = {
+            val pack = onColorTextForceUpdate()
+            setActiveBox(floatArrayOf(pack.hue, pack.saturation, pack.value))
+        }
+        previousBox.setOnClickListener {
+            val hsv = FloatArray(3)
+            Color.colorToHSV(previousColor.toArgb(), hsv)
+            //used to set the color of the active box to the previous color
+            setActiveBox(hsv)
+        }
+        submit.setOnClickListener {
+            onSubmission(
+                Color.HSVToColor(
+                    floatArrayOf(
+                        slider.hue,
+                        square.sat,
+                        square.valu
+                    )
+                ).toColor()
+            )
+            sheet.dismiss()
+        }
+        cancel.setOnClickListener {
+            sheet.dismiss()
+        }
+
+        slider.onHueChange = {
+            square.hue = it
+            square.invalidate()
+            onHueUpdate(it)
+            updateForceViews()
+        }
+        square.onSatOrValChange = { saturation, value ->
+            onSatOrValUpdate(saturation, value)
+            square.sat = saturation
+            square.valu = value
+            updateForceViews()
+        }
+        square.onDown = onViewTouch
+        square.onUp = onViewLift
+        slider.onDown = onViewTouch
+        slider.onUp = onViewLift
+
+        hView.addTextChangedListener {
+            val passedInt =
+                hView.text.toString().toFloatOrNull()?.let { string ->
+                    max(min(360f, string), 0f)
+                }
+            passedInt?.let { hue ->
+                slider.sliderSetHue(hue)
+                onHueUpdate(hue)
+                updateForceViews()
+            }
+        }
+        sView.addTextChangedListener {
+            val saturation = sView.text.toString().toFloatOrNull()
+                ?.let { string -> max(min(1f, string.div(100)), 0f) }
+            val value = vView.text.toString().toFloatOrNull()
+                ?.let { string -> max(min(1f, string.div(100)), 0f) }
+            saturation?.let { sat ->
+                value?.let { valu ->
+                    onSatOrValUpdate(sat, valu)
```

Comment:
good idea to ignore the hue here, I didn't do that and was inefficient and was assigning a hue value when I didn't need to do it

👍